types/event_record.go

```diff
@@ -72,39 +78,116 @@ type EventBalancesTransfer struct {
+// when an existing index is reassigned to another AccountId
 type EventIndicesNewAccountIndex struct {
 	Phase        Phase
 	AccountID    AccountID
 	AccountIndex AccountIndex
 	Topics       []Hash
 }
-type EventBalancesNewAccount struct {
+
+// EventOffencesOffence is emitted when there is an offence reported of the given kind happened at the session_index
+// and (kind-specific) time slot. This event is not deposited for duplicate slashes
+type EventOffencesOffence struct {
+	Phase          Phase
+	Kind           Bytes16
+	OpaqueTimeSlot Bytes
+	Topics         []Hash
+}
+
+// EventSessionNewSession is emitted when a new session has happened. Note that the argument is the session index,
+// not the block number as the type might suggest
+type EventSessionNewSession struct {
+	Phase        Phase
+	SessionIndex U32
+	Topics       []Hash
+}
+
+// EventStakingOldSlashingReportDiscarded is emitted when an old slashing report from a prior era was discarded because
+// it could not be processed
+type EventStakingOldSlashingReportDiscarded struct {
+	Phase        Phase
+	SessionIndex U32
+	Topics       []Hash
+}
+
+// EventStakingReward is emitted when all validators have been rewarded by the given balance
+type EventStakingReward struct {
+	Phase   Phase
+	Balance U128
+	Topics  []Hash
+}
+
+// EventStakingSlash is emitted when one validator (and its nominators) has been slashed by the given amount
+type EventStakingSlash struct {
 	Phase     Phase
 	AccountID AccountID
 	Balance   U128
 	Topics    []Hash
 }
-type EventBalancesReapedAccount struct {
-	Phase     Phase
-	AccountID AccountID
-	Topics    []Hash
+
+// EventSystemExtrinsicSuccess is emitted when an extrinsic completed successfully
+type EventSystemExtrinsicSuccess struct {
+	Phase  Phase
+	Topics []Hash
 }
-type EventSessionNewSession struct {
-	Phase        Phase
-	SessionIndex U32
-	Topics       []Hash
+
+// EventSystemExtrinsicFailed is emitted when an extrinsic failed
+type EventSystemExtrinsicFailed struct {
+	Phase         Phase
+	DispatchError DispatchError
+	Topics        []Hash
 }
 
 // EventRecords is a default set of possible event records that can be used as a target for
 // `func (e EventRecordsRaw) Decode(...`
 type EventRecords struct {
-	System_ExtrinsicSuccess []EventSystemExtrinsicSuccess //nolint:stylecheck,golint
-	System_ExtrinsicFailed  []EventSystemExtrinsicFailed  //nolint:stylecheck,golint
-	Indices_NewAccountIndex []EventIndicesNewAccountIndex //nolint:stylecheck,golint
-	Balances_NewAccount     []EventBalancesNewAccount     //nolint:stylecheck,golint
-	Balances_ReapedAccount  []EventBalancesReapedAccount  //nolint:stylecheck,golint
-	Balances_Transfer       []EventBalancesTransfer       //nolint:stylecheck,golint
-	Session_NewSession      []EventSessionNewSession      //nolint:stylecheck,golint
+	Balances_NewAccount                []EventBalancesNewAccount                //nolint:stylecheck,golint
```

Comment:
Are there any other events that we should keep an eye on? Can you point me to the substrate code where you found all these?

Reply:
They are declared in each module in `decl_event` of the SRML, e. g. https://github.com/paritytech/substrate/blob/master/srml/balances/src/lib.rs#L244 . 

A secondary, curated list can be found here: https://polkadot.js.org/api/substrate/events.html Not that this is not exhaustive and can be outdated.

I added a sane default here, but decoding will fail if not all events that occur in a block are declared – so we should add all of them at some point.